Scripts/SL-ECFAHelpers.lua

```diff
@@ -0,0 +1,287 @@
+end
+
+-- Calculate the max DP using the formula above, scaled so that the maximum
+-- DP available from any one chart in the event is capped at 1000
+-- (i.e. set a 14 that maxes out the radar at 1000)
+local theAngriestBoi = {
+	rating = 14,
+	speed = 10,
+	stamina = 10,
+	tech = 10,
+	movement = 10,
+	timing = 10,
+	rhythms = 10,
+	gimmick = 3
+}
+local theDPScalar = 10000.0 / CalculateMaxDP_Unscaled(theAngriestBoi)
+
+CalculateMaxDPByTechRadar = function(techRadarTable)
+	-- ECFA: Calculate the maximum DP available for the chart with the
+	-- tech radar values in the table provided.
+	-- 	
+	-- Parse the #CHARTSTYLE: field to get radar values.
+	-- e.g., {speed = 5, stamina = 6, tech = 7, movement = 10, timing = 9, gimmick = 0}
+	
+	if not techRadarTable then return nil end
+
+	-- Scan for all required parameters.
+	local requiredParams = {'speed', 'stamina', 'tech', 'movement', 'rhythms', 'gimmick', 'rating'}
+	for _, v in ipairs(requiredParams) do
+		if techRadarTable[v] == nil then
+			return nil
+		end
+	end
+
+	-- Calculate the max DP.
+	return CalculateMaxDP_Unscaled(techRadarTable) * theDPScalar
+end
+
+SongNameDuringSet = function(self, item)
+	-- NOTE: For this function to get called at the proper time, the following
+	-- lines must be updated in metrics.ini:
+	--
+	-- [MusicWheelItem]
+	-- SongNameSetCommand=%function(self, item) SongNameDuringSet(self, item) end
+	
+	-- (from original SongNameSetCommand)
+	-- hack to recolor song titles back EVERY SetCommand (i.e. a lot)
+	self:diffuse(Color.White)
+
+	-- ECFA: Change song title to include the Challenge chart's rating.
+	if item.Song then
+		-- only do this for songs in an ECFA 2021 folder
+		if not item.Song:GetGroupName():find("ECFA 2021") then return end
+
+		-- Grab a list of all steps for the current mode.
+		local allSteps = item.Song:GetStepsByStepsType(GAMESTATE:GetCurrentStyle():GetStepsType())
+
+		-- Find the chart occupying the highest non-Edit slot.
+		local highestRegularDiff = nil
+		local highestRegularChart = nil
+		local techRadar
+		for _, diff in ipairs({
+			'Difficulty_Beginner',
+			'Difficulty_Easy',
+			'Difficulty_Medium',
+			'Difficulty_Hard',
+			'Difficulty_Challenge'
+		}) do
+			for _, step in ipairs(allSteps) do
+				techRadar = ParseTechRadar(step:GetChartStyle())
+				if (step:GetDifficulty() == diff) and techRadar then
+					highestRegularChart = step
+					highestRegularDiff = diff
+				end
+			end
+		end
+		
+		-- Get the block rating / tech max associated with that chart
+		-- and append it to the title.
+		if highestRegularChart then
+			local blockRating = tonumber(highestRegularChart:GetMeter())
+			techRadar = ParseTechRadar(highestRegularChart:GetChartStyle())
+			techRadar.rating = blockRating
+			local techMaxDP = math.floor(CalculateMaxDPByTechRadar(techRadar))
+
+			-- Title gets a prepended block rating
+			-- Subtitle gets either a Cmod directive, a max DP calculation, or
+			-- both, separated by a pipe (e.g. "573 DP | No Cmod")

```

Comment:
It appears this will duplicate your no cmod item on the wheel, maybe it should be removed from there, or here

Reply:
Do we have a no cmod song to test this with?

Reply:
sent